clients/buckets/bucket.go

```diff
@@ -0,0 +1,569 @@
+// @license
```

Comment:
Note: This is functionally the same `api/client/buckets` client as before, but direct calls to the rest.Client are replaced with using the newly introduced "plain API CRUD" bucket client.

(automation for now is just moved without introducing the plain API client, so we can review the concept on one sample)